tests/python/unittest/test_tir_transform_regenerate_def.py

```diff
@@ -0,0 +1,94 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pytest
+import sys
+
+import tvm
+from tvm.script import tir as T
+from tvm.tir.function import PrimFunc
+from tvm.tir.stmt import Block
+
+
+def _check_func_signature(lhs: PrimFunc, rhs: PrimFunc):
+    for x, y in zip(lhs.params, rhs.params):
+        assert x != y
+        assert lhs.buffer_map[x] != rhs.buffer_map[y]
+
+
+def _check_block_signature(lhs: Block, rhs: Block):
+    for x, y in zip(lhs.iter_vars, rhs.iter_vars):
+        assert x != y
+    for x, y in zip(lhs.alloc_buffers, rhs.alloc_buffers):
+        assert x != y
+    for x, y in zip(lhs.match_buffers, rhs.match_buffers):
+        assert x != y
```

Comment:
This check is too simple. 
we also need to check:
1. read and write signature are correctly remapped.
2. vars in buffer's shape and stride are correctly remapped
3. vars and buffer in MatchBufferRegion are correctly remapped